### PR TITLE
MNT fix a few pandas warnings

### PR DESCRIPTION
**Pull request by adrinjalali**
A few fixes to improve pandas compatibility with latest pandas release.

I am very confused with the failure on the `TestEqualizedOdds.test_many_sensitive_feature_groups_warning[True-ensure_dataframe-ensure_dataframe-ensure_dataframe]` test though.

Maybe @riedgar-ms would know?

We also need to accept `labels` now in `MetricFrame`. I tried to fix that, but the metricframe codebase is very confusing to me.

### CODE REVIEW

File: requirements.txt
Comment (adrinjalali) [inline requirements.txt:1]: no idea why this is popping up here. This is the same as what's in `main`.

**Comment by riedgar-ms** [comment]
TestEqualizedOdds is more a @MiroDudik  question.

File: fairlearn/metrics/_disaggregated_result.py [hunk 1]
```diff
@@ -383,7 +385,9 @@ def create(
             all_grouping_names = control_feature_names + all_grouping_names
 
         temp = data.groupby(all_grouping_names).apply(
-            apply_to_dataframe, metric_functions=annotated_functions
+            apply_to_dataframe,
+            metric_functions=annotated_functions,
+            include_groups=False,
```
[the remark this answers is not in the document]
Comment (adrinjalali) [reply fairlearn/metrics/_disaggregated_result.py:390]: You're looking at the wrong page: https://pandas.pydata.org/docs/reference/api/pandas.core.groupby.DataFrameGroupBy.apply.html

**Comment by adrinjalali** [comment]
This PR:

```
= 108 failed, 6219 passed, 322 skipped, 1080 deselected, 4 xfailed, 4 warnings in 666.24s (0:11:06) =
```

`main` last run:

```
= 2925 failed, 3363 passed, 322 skipped, 1080 deselected, 4 xfailed, 4 warnings, 39 errors in 657.80s (0:10:57) =
```

File: fairlearn/metrics/_disaggregated_result.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (riedgar-ms) [reply fairlearn/metrics/_disaggregated_result.py:390]: Got it, thanks.

**Review by romanlutz — APPROVED** [review]

**Pull request merged**